doc/Container Hardening/SONiC_container_hardening_HLD.md

```diff
@@ -0,0 +1,349 @@
+					--tmpfs /var/tmp \
+					--env "NAMESPACE_ID"="$DEV" \
+					--env "NAMESPACE_PREFIX"="$NAMESPACE_PREFIX" \
+					--env "NAMESPACE_COUNT"=$NUM_ASIC \
+					--name=database_no_net \
+					--cap-drop=NET_ADMIN \
+					docker-database:latest
+
+To disable the sharing of the networking stack between the host and a container we need to remove the flag: `--net=host`. Because we have not specified any `--network` flag, the containers connect to the default bridge network `--net=bridge`.
+To support port forwarding we are required to add the flag:  `-p <port>:<port>`
+
+The "new" docker creation file database.sh can be seen in the code block below:
+
+Docker with port forwarding and default bridge network
+
+	docker create --privileged -t -v /etc/sonic:/etc/sonic:ro  \
+					**-p 6379:6379** \
+					-e RUNTIME_OWNER=local \
+					--uts=host \
+					--log-opt max-size=2M --log-opt max-file=5 \
+					--tmpfs /tmp \
+					$DB_OPT \
+					$REDIS_MNT \
+					-v /usr/share/sonic/device/$PLATFORM:/usr/share/sonic/platform:ro \
+					--tmpfs /var/tmp \
+					--env "NAMESPACE_ID"="$DEV" \
+					--env "NAMESPACE_PREFIX"="$NAMESPACE_PREFIX" \
+					--env "NAMESPACE_COUNT"=$NUM_ASIC \
+					--name=$DOCKERNAME \
+					docker-database:latest \
+
+**How we did it?**
+
+To create a docker with the flags above it is required to set the "new" flag in the file docker_image_ctl.js. Follow the call `docker create {{docker_image_run_opt}} \`: 
+and replace the `–--net=$NET`.
+docker flag generation
+
+	{%- if docker_container_name != "database" %}
+					--net=$NET \
+	{%- endif %}
+	{%- if docker_container_name == "database" %}
+					-p 6379:6379 \
+	{%- endif %}
+
+#### How to check?
+
+Go into the docker - `docker exec -it docker bash`
+Run `ifconfig`.
+
+On a docker with host network - you'll be able to view all physical interfaces.
+On a docker without host network - we'll see only eth0 and lo.
+
+## 5. SAI API
+
+N/A
+
+## 6. Configuration and management
+
+N/A - no configuration management/changes are required.
+
+### 6.1. Manifest (if the feature is an Application Extension)
+
+N/A
+
+### 6.2. CLI/YANG model Enhancements
+
+N/A
+We are not adding CLI commands or management capabilities to the system with this item.
+
+### 6.3. Config DB Enhancements
+
+N/A - DB should remain the same
+
+## 7. Warmboot and Fastboot Design Impact
+
+No impact on all boot sequences, as this item should be seemlessly integrated into the system and achieve the same functionality level as before.
+
+## 8. Restrictions/Limitations
+
+## 9. Testing Requirements/Design
+
+To define this item completed - we are required to run the full CI and check that nothing has been broken from the changes proposed in this HLD.
+In addition - we should test that the mitigations are applicable for the relevant containers.
+
+### 9.1 Unit Test cases
+
+N/A, this feature will be checked on a system level.
+
+### 9.2 System Test cases
+
```

Comment:
Is there any impact w.r.t memory and cpu as it is running in different network name spaces?

Reply:
Negligible, as it is part of the infra.
I'll be more precise - the impact is as much of how much the Linux ACL takes, but this should not be visible to any user.